src/Microsoft.Health.Dicom.Api/Features/Audit/AuditLoggingFilterAttribute.cs

```diff
@@ -27,6 +28,7 @@ public override void OnActionExecuted(ActionExecutedContext context)
 
             if (context.Exception != null)
             {
+                context.HttpContext.Response.StatusCode = (int)ExceptionHandlingHelper.GetStatusCode(context.Exception);
```

Comment:
I am curious to understand why the AuditFilter is setting the response statusCode.
Is it not going through the middleware on the way out and getting set there?

Reply:
That is partially correct. The way DICOM handles exception, ExceptionHandlingMiddleware is responsible for setting the correct status code. However, this middleware is called after OnActionExecuted is called in the AuditFilter which means that when OnActionExecuted is called, the status code is still set to OK (200).

Now this is different from FHIR because FHIR uses a class level annotation of OperationOutcomeExceptionFilterAttribute because of which OnResultExecuted is called in the AuditFilter. Since DICOM's layout is different and does not have this class level attribute, OnResultExecuted is not called in the cases of Exceptions, hence, we have to rely on OnActionExecuted.

Reply:
Have you looked at creating a AudiLoggerMiddleware that gets called after ExceptionHandlingMiddleWare on the way out, that can extract the claim and log the the context?

Looks like for auditing we need to capture the response right before it leaves instead of here